internal/api/httpsrv/handler.go

```diff
@@ -104,7 +105,10 @@ func (h *APIHandler) Routes(rg *echo.Group) {
 
 func (h *apiHandler) rollbackAndReturnError(ctx context.Context, httpcode int, msg string) *echo.HTTPError {
 	if err := h.engine.RollbackContext(ctx); err != nil {
-		return echo.NewHTTPError(http.StatusBadGateway, err)
+		return echo.NewHTTPError(
+			http.StatusInternalServerError,
+			fmt.Errorf("%w: %w", ErrDBRollbackFailed, err),
+		)
```

Comment:
What do you think of something maybe more like:

```suggestion
		return echo.NewHTTPError(
			http.StatusInternalServerError,
			fmt.Errorf("%s and %w", msg, ErrDBRollbackFailed),
		).SetInternal(err)
```

Which would produce an error in the response of `failed to create group and failed to rollback database transaction`

This returns a clean message, and keeps the more detailed error internal for logs.